src/sidebar/components/MenuItem.js

```diff
@@ -140,44 +173,114 @@ export default function MenuItem({
         }
     }
   };
+
+  let renderedIcon = null;
+  if (icon && icon !== 'blank') {
+    renderedIcon = iconIsUrl ? (
+      <img className="w-4 h-4" alt={iconAlt} src={icon} />
+    ) : (
+      <Icon name={icon} classes="h-3 w-3" />
+    );
+  }
+  const leftIcon = isSubmenuItem ? null : renderedIcon;
+  const rightIcon = isSubmenuItem ? renderedIcon : null;
+
+  // MenuItem content layout consists of:
+  // - Sometimes a left item, which may contain an icon or serve as
+  //   an indenting space for label alignment
+  // - Always a label
+  // - Sometimes a right item, which contains an icon (submenu items)
+  // - Sometimes a submenu-toggle control (only if the item has a submenu)
+  const hasLeftItem = leftIcon || isSubmenuItem || icon === 'blank';
+  const hasRightItem = rightIcon && isSubmenuItem;
+
+  const menuItemContent = (
+    <>
+      {hasLeftItem && (
+        <div
+          className="w-7 flex items-center justify-center"
+          data-testid="left-item-container"
+        >
+          {leftIcon}
+        </div>
+      )}
+      <span className="flex items-center grow whitespace-nowrap px-1">
+        {label}
+      </span>
+      {hasRightItem && (
+        <div
+          className="w-8 flex items-center justify-center"
+          data-testid="right-item-container"
+        >
+          {rightIcon}
+        </div>
+      )}
+      {hasSubmenuVisible && (
+        <SubmenuToggle
+          title={`Show actions for ${label}`}
+          isExpanded={isSubmenuVisible}
+          onToggleSubmenu={onToggleSubmenu}
+        />
+      )}
+    </>
+  );
+
+  const wrapperClasses = classnames(
+    'hyp-u-outline-on-keyboard-focus--inset',
+    'w-full min-w-[150px] flex items-center select-none',
+    'border-b',
+    // Set this container as a "group" so that children may style based on its
+    // layout state.
+    // See https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state
+    'group',
+    {
+      'min-h-[30px] font-normal': isSubmenuItem,
+      'min-h-[40px] font-medium': !isSubmenuItem,
+      'bg-grey-1 hover:bg-grey-3': isSubmenuItem || isExpanded,
+      'bg-white hover:bg-grey-1': !isSubmenuItem && !isExpanded,
+      // visual "padding" on the right is part of SubmenuToggle when rendered,
+      // but when not rendering a SubmenuToggle, we need to add some padding here
+      'pr-1': !hasSubmenuVisible,
+    },
+    {
+      // When the item is selected, show a left border to indicate it
+      'border-l-[4px] border-l-brand': isSelected,
+      // Add equivalent padding to border size when not selected. This instead
+      // of a transparent left border to make focus ring cover the full
+      // menu item. Otherwise the focus ring will be inset on the left too far.
+      'pl-1': !isSelected,
```

Comment:
Is there a situation when `pl-1` could end up being a value other than `4px` and thus inconsistent with the border above?

Reply:
This is a really insightful question. In practice, almost assuredly not, but in theory, yes. I'd like to give this a little more thought.

Reply:
I'm opting to change this to `pl-[4px]` because the pixel-specificity is relevant here. Thanks for noting this.